### PR TITLE
kn source-list types: adding column to specify built-in source

### PR DESCRIPTION
**Pull request by eletonia**
## Description
`kn source list-types` currently shows the `TYPE`, `NAME`, and `DESCRIPTION` of available sources. 
This PR adds a `S` column which shows whether the source is built-in or not:
```
$ kn source list-types
TYPE              S     NAME                                   DESCRIPTION
ApiServerSource   X     apiserversources.sources.knative.dev   Watch and send Kubernetes API events to addressable
ContainerSource   X     containersources.sources.knative.dev   Generate events by Container image and send to addressable
PingSource        X     pingsources.sources.knative.dev        Send periodically ping events to addressable
SinkBinding       X     sinkbindings.sources.knative.dev       Binding for connecting a PodSpecable to addressable
```

## Changes
* changed `human_readable_flags.go` to add the new `S` column
* `list_test.go` now checks for `S` column and appropriate values
* `source_list_test.go` e2e test now checks for `S` column

## Reference

Fixes a part of #633 More discussion is needed on how to handle plugin sources.

### CODE REVIEW

**Comment by eletonia** [comment]
@googlebot I fixed it.

**Comment by dsimansk** [comment]
@eletonia thanks for the contribution!

The change looks good at a first glance. However, google-cla bot is confused due to mix of 2 accounts adding commit. I'd better if you could all those commits into one under your preferred account.

You can check how to fill the licence [agreement](https://knative.dev/community/contributing/contributing/#contributor-license-agreements) for the future contributions.

/ok-to-test

File: pkg/kn/commands/source/human_readable_flags.go [hunk 1]
```diff
@@ -37,12 +37,21 @@ var sourceTypeDescription = map[string]string{
...
+	"KafkaSource":     "NO",
+}
+
 // ListTypesHandlers handles printing human readable table for `kn source list-types`
 func ListTypesHandlers(h printers.PrintHandler) {
 	sourceTypesColumnDefinitions := []metav1beta1.TableColumnDefinition{
 		{Name: "Type", Type: "string", Description: "Kind / Type of the source type", Priority: 1},
 		{Name: "Name", Type: "string", Description: "Name of the source type", Priority: 1},
 		{Name: "Description", Type: "string", Description: "Description of the source type", Priority: 1},
+		{Name: "Built-In Source", Type: "string", Description: "Built-in source", Priority: 1},
```
Comment (dsimansk) [inline pkg/kn/commands/source/human_readable_flags.go:54]: I'd shorten the name to `Built-In` due to the fact that we have only very short `YES, NO` values. 

And my preferred order would be `Type, Name, Built-In, Description` to allow more room for various description length.

Comment (eletonia) [reply pkg/kn/commands/source/human_readable_flags.go]: Thanks for the feedback @dsimansk 
I shortened it to `Built-In` and moved description to the end in this [commit](https://github.com/knative/client/pull/1246/commits/3e545e11342f91de0aa895456eb003e76bd248eb) 
Now the output looks like this:
```bash-5.1$ ./kn source list-types
TYPE              NAME                                   BUILT-IN   DESCRIPTION
ApiServerSource   apiserversources.sources.knative.dev   YES        Watch and send Kubernetes API events to addressable
ContainerSource   containersources.sources.knative.dev   YES        Generate events by Container image and send to addressable
PingSource        pingsources.sources.knative.dev        YES        Send periodically ping events to addressable
SinkBinding       sinkbindings.sources.knative.dev       YES        Binding for connecting a PodSpecable to addressable
```

I fixed the CLA issue by amending those commits with the correct email that I used to sign the CLA

File: CHANGELOG.adoc [hunk 2]
```diff
@@ -12,6 +12,18 @@
...
 
+## v0.22.0 (unreleased)
+
+[cols="1,10,3", options="header", width="100%"]
+|===
+| | Description | PR
+
+| 🎁
+| Add `BUILT-IN` column to `kn source list-types`
+| https://github.com/knative/client/pull/[#1246]
```
Comment (dsimansk) [inline CHANGELOG.adoc:23]:
```suggestion
| https://github.com/knative/client/pull/1246[#1246]
```

**Review by dsimansk — APPROVED** [review]
/lgtm

**Review by maximilien — COMMENTED** [review]
Thanks for the contribution. The question is whether this could be done dynamically since as new sources are added and introduced you’ll need to change your hard coded list.

**Comment by eletonia** [comment]
Details for `apiserver`, `container`, `ping`, and `binding` are hardcoded in other [parts](https://github.com/eletonia/client/blob/efa9c196a6ff2f112229aab86c65fe4cf218196b/pkg/kn/commands/source/human_readable_flags.go#L29) of the code since these are the available subcommands for for `kn source`:
```
$ kn source --help
Manage event sources

Usage:
  kn source SOURCE|COMMAND [options]

Aliases: 
  source, sources

Available Commands:
  apiserver   Manage Kubernetes api-server sources
  binding     Manage sink bindings
  container   Manage container sources
  list        List event sources
  list-types  List event source types
  ping        Manage ping sources
```

In a future issue with plugin support, we can dynamically display the description and source type if it is a plugin as suggested in the [comments of this code ](https://github.com/eletonia/client/blob/efa9c196a6ff2f112229aab86c65fe4cf218196b/pkg/kn/commands/source/human_readable_flags.go#L34)

**Comment by itsmurugappan** [comment]
Thanks for your contribution, couple of things, 
* should the values of this column be "Yes/No" or "BuiltIn/Plugin" ? 
* to keep it simple, anything apart from the built in source could be displayed as plugin (for kafkasource, githubsource etc)

**Comment by dsimansk** [comment]
>     * should the values of this column be "Yes/No" or "BuiltIn/Plugin" ?
>     * to keep it simple, anything apart from the built in source could be displayed as plugin (for kafkasource, githubsource etc)

Well, that'd imply easily that such a plugin actually exists to support the source, but it's not necessary true for a lot of eventing sources.

**Comment by rhuss** [comment]
Thanks for the PR, looks good in general, but I would be denser to save screen real estate. Also, I think that for the user it is only important whether a source is supported by the client or not, regardless of whether it is built-in or supported via a plugin. 

E.g. instead of YES / NO, just use an "X" (and nothing if not builtin), and instead of "BUILT-IN" header, maybe just an "S" letter (or something else), or even no header. Then, I would put that in the second or third column, before the description because this column is of fixed length (ideally only 1 char), while a description can be long and varying in length. In general, the longest possible column should be last because this can be the only one that possible needs not to be truncated when being longer than a certain limit.

**Comment by dsimansk** [comment]
Ahh, so there's `prettier-exclude.yaml` good to know. Thanks! :)

**Comment by eletonia** [comment]
> All (the pull request submitter and all commit authors) CLAs are signed, **but** one or more commits were authored or co-authored by someone other than the pull request submitter.
> 
> We need to confirm that all authors are ok with their commits being contributed to this project. Please have them confirm that by leaving a comment that contains only `@googlebot I consent.` in this pull request.
> 
> _Note to project maintainer:_ There may be cases where the author cannot leave a comment, or the comment is not properly detected as consent. In those cases, you can manually confirm consent of the commit author(s), and set the `cla` label to `yes` (if enabled on your project).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fknative%2Fclient%2Fpull%2F1246) for more info**.

@itsmurugappan @dsimansk I think this is because I included your latest commits in this PR

**Comment by eletonia** [comment]
@rhuss Thanks for the feedback! I changed the `BUILT-IN` column to `S` and the values to `X` if they are built-in sources. This column is now the second column: 
```
$ ./kn source list-types
TYPE              S     NAME                                   DESCRIPTION
ApiServerSource   X     apiserversources.sources.knative.dev   Watch and send Kubernetes API events to addressable
ContainerSource   X     containersources.sources.knative.dev   Generate events by Container image and send to addressable
PingSource        X     pingsources.sources.knative.dev        Send periodically ping events to addressable
SinkBinding       X     sinkbindings.sources.knative.dev       Binding for connecting a PodSpecable to addressable
```

**Comment by dsimansk** [comment]
@eletonia something went wrong with the last commit to fix changelog conflict.

Could you please try the following?

```bash
# Remove the very last commit
git reset --hard HEAD^

# Edit the changelog formatting and add as a fresh commit
git add .
git commit -m 

# Force push to your branch
git push -f
```

**Comment by eletonia** [comment]
@dsimansk I think I fixed the merge conflict. The CLA bot is still failing though

**Comment by dsimansk** [comment]
@googlebot I consent.

**Comment by dsimansk** [comment]
@eletonia at least we know it's not my "consent". :)

**Comment by eletonia** [comment]
Thanks @dsimansk There are no other authors that need to consent. It may have to be manually changed 

> Note to project maintainer: There may be cases where the author cannot leave a comment, or the comment is not properly detected as consent. In those cases, you can manually confirm consent of the commit author(s), and set the cla label to yes (if enabled on your project).

**Comment by dsimansk** [comment]
@itsmurugappan could you please try to comment with `@googlebot I consent.` I think, the merge commit might be confusing cla bot as it brings your changelog update from `master` branch. 

@eletonia I believe for the manual confirmation it needs to be also Googler. It might be quicker to just create a fresh PR. @rhuss any suggestion would be welcome pls?

**Comment by itsmurugappan** [comment]
@googlebot I consent.

**Comment by rhuss** [comment]
@eletonia @dsimansk the best way is to locally squash everything to a local commit and then do a forced push. Then the googlebot is happy again :)

Alternatively you could of course do a fresh PR but I think the first option might be easier ?

**Comment by eletonia** [comment]
@dsimansk @rhuss Thanks for the suggestion! I squashed my commits and looks like all the tests are passing now

**Comment by dsimansk** [comment]
Thanks @eletonia! I think it's all good now.
/approve
/lgtm

**Pull request merged**